protocols/dcutr/src/handler/relayed.rs

```
@@ -148,12 +148,11 @@ impl Handler {
 
     fn on_listen_upgrade_error(
         &mut self,
-        ListenUpgradeError { error, .. }: ListenUpgradeError<
+        ListenUpgradeError { .. }: ListenUpgradeError<
             (),
             <Self as ConnectionHandler>::InboundProtocol,
         >,
     ) {
-        libp2p_core::util::unreachable(error.into_inner());
```

Why is this statement removed?

---

Hi Elena! Glad you chime in, cause it's giving unreachable warning lints, what's the advantage in keeping it? (and the others)

---

hey guys, because of the rustup update on CI to use beta channel,  on my [recent PR](https://github.com/libp2p/rust-libp2p/pull/6190/files#diff-ec8e551b52b7a5f7d115f49f3c4e75c8dab2929ccd8f66f060204e34b2b3f69cR156) I had the same issues.

Is it adequate to just use `#[allow(unreachable_code)]` without removing the whole line?

Beacuse ultimately, it's just the variant of `StreamUpgradeError::Apply` which is `Infallible`.

---

Discussed out-of-band. Removing the line is fine, since `libp2p_core::util::unreachable` also just silently discards the error.